src/internal/connector/onedrive/permission.go

```diff
@@ -163,6 +183,15 @@ func diffPermissions(before, after []UserPermission) ([]UserPermission, []UserPe
 		}
 	}
 
+	// Since we only try to restore permissions if the mode is
+	// SharingModeCustom, if all the permission match, we have to
+	// delete and recreate at least one to make sure that the
+	// permission inheritance chain is broken.
+	if len(added) == 0 && len(removed) == 0 {
+		added = append(added, before[0])
+		removed = append(removed, before[0])
+	}
```

Comment:
It is sensitive to ordering. I would have liked to move this logic to the [function](https://github.com/alcionai/corso/blob/762f7bc69efa59ba1632b49516fa7d68f031801f/src/internal/connector/onedrive/permission.go#L355) that does the restore, but that function only has the diffed version. That said, the function that does the restore already relies on removing before adding. Let me add a comment on both though.

Reply:
I did some more digging and realized that this can be a problem is the previous one has empty permissions.

Reply:
On second though, let me remove this completely for now. It is a much deeper rabbit hole with not as much importance IMO. I can't think of a straight forward way to do this as we might have to ideally track if individual permissions are inherited or custom for which data is not even available AFAIC. There is `inheritedFrom` property, but it never shows up for any permission entries that are for items with custom permissions. I've created another issue to track this at https://github.com/alcionai/corso/issues/3126

Let me also loop in @gmatev  in case he has any thoughts on this.

Reply:
We can always make it not inherit by deleting and recreating common ones. We were previously relying on ids to do this, but as we have seen, that is not reliable.